src/jmespath.net.parser/Utils/PushbackQueue.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DevLab.JmesPath.Utils
+{
+    /// <summary>
+    /// PushbackQueue provides the infrastructure for pushing back token 
+    /// information to a host scanner. The class provides an API for 
+    /// initializing new lookahead buffers and adding those to the queue.
+    /// The host scanner must be modified so that yylex takes tokens from
+    /// the queue while the queue is not empty.
+    /// </summary>
+    public class PushbackQueue<Obj>
```

Comment:
Does the class really need to be public ?

Reply:
Made internal in latest checkin